### PR TITLE
Use overriden double/float GetHashCode for default struct hashcodes

### LINKED ISSUE #16545: Default hashcode for struct with double or float field is wrong
Repro:

```
using System;

struct StructWithDoubleField
{
    public double value;
}

class My {

    static void Main() {
        StructWithDoubleField s1;
        s1.value = 0.0;
        StructWithDoubleField s2;
        s2.value = -0.0;

        Console.WriteLine(s1.GetHashCode() == s2.GetHashCode() ? "PASSED" : "FAILED");
    }

}
```

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #16545

### CODE REVIEW

**Comment by jkotas** [comment]
cc @MichalStrehovsky

File: src/vm/comutilnative.cpp [hunk 1]
```diff
@@ -2087,60 +2088,65 @@ static INT32 RegularGetValueTypeHashCode(MethodTable *mt, void *pObjRef)
...
-            for (INT32 i = 0; i < count; i++)
+            _ASSERTE(!field->IsRVA());
+            if (field->IsObjRef())
             {
-                FieldDesc *field = fdIterator.Next();
-                _ASSERTE(!field->IsRVA());
-                void *pFieldValue = (BYTE *)pObj + field->GetOffsetUnsafe();
-                if (field->IsObjRef())
+                // if we get an object reference we get the hash code out of that
+                if (*(Object**)((BYTE *)pObjRef + field->GetOffsetUnsafe()) != NULL)
```
Comment (janvorli) [inline src/vm/comutilnative.cpp]: I wonder what was the reason for removing the pFieldValue and using the `(BYTE *)pObjRef + field->GetOffsetUnsafe()` at all places.

Comment (jkotas) [reply src/vm/comutilnative.cpp]: I would have to add more GC protects for the pFieldValue. The original code tried to step around the GC protect, but it is not easy to do in the extra code that I have added (and I am also not sure whether the original got it 100% right).

**Comment by MichalStrehovsky** [comment]
Thanks! I'm glad I don't have to replicate this bug on the CoreRT side :)

**Pull request merged**